hypothesis-python/src/hypothesis/utils/conventions.py

```diff
@@ -19,9 +19,6 @@ def __repr__(self):
         return self.identifier
 
 
-class InferType(UniqueIdentifier):
-    """We have a subclass for `infer` so we can type-hint public APIs."""
-
-
-infer = InferType("infer")
+infer = ...
+InferType = type(Ellipsis)
```

Comment:
https://docs.python.org/3/library/types.html#types.EllipsisType if we can, for the sake of typecheckers, and fall back to this on <=3.9

Reply:
My understanding is that, because we can't use `TypeAlias` we'll to define:

```python
if <is py310>:
    InferType = typing.EllipsisType
else:
    InferType = type(Ellipsis)
```

at the top of each module where `InferType` is needed. Does that match what you are envisioning?

Reply:
```python
try:
    InferType = types.EllipsisType
except AttributeError:  # pragma: no branch
    InferType = type(Ellipsis)
```

`types` not `typing`, and I think if we do it this way Mypy will think we're in the first branch on older versions too... which is *morally* true at any rate.

Reply:
> `types` not `typing`

heh.. that was pure muscle memory

Reply:
> ```python
> try:
>    InferType = types.EllipsisType
> except AttributeError:  # pragma: no branch
>    InferType = type(Ellipsis)
>```

Are you sure we want to do this vs a version check? Both [mypy and pyright support the latter]. For the try-except block on python < 3.10, pyright infers this as `InferType: Unknown | Type[ellipsis]`

Reply:
Sure, let's go with a version check then.